index.js

```diff
@@ -48,50 +48,274 @@ const player = {
     { id: 5, name: 'Israeli', songs: [4, 5] },
   ],
   playSong(song) {
-    console.log(/* your code here */)
+    return `Playing ${song.title} from ${song.album} by ${song.artist} | ${convertSecondsToMinutes(song.duration)}.`
   },
 }
 
 function playSong(id) {
-  // your code here
+  let song = getSongById(id)
+  console.log(player.playSong(song))
 }
 
 function removeSong(id) {
-  // your code here
+  //remove song from songs
+  player.songs.splice(getSongIndexById(id), 1)
+
+  // remove from playlists
+  for (let playlist of player.playlists) {
+    for (let songID of playlist.songs) {
+      if (songID === id) {
+        playlist.songs.splice(playlist.songs.indexOf(id), 1)
```

Comment:
This is not the most efficient way to implement this, since `indexOf` effectively does another walkthrough of the array.
There are several ways to solve this, which are also elegant, but I'll let you enjoy thinking about them yourself :)

Reply:
Fixed it to be more efficient. Thanks!